lib/twom.h

```diff
@@ -0,0 +1,162 @@
+/* twom.h - twoskip implementation with MVCC capability
+ *
+ * Copyright (c) 2025 Fastmail Pty Ltd
+ *
+ * https://creativecommons.org/publicdomain/zero/1.0/
+ *
+ *   The person who associated a work with this deed has dedicated the work to the
+ *   public domain by waiving all of his or her rights to the work worldwide under
+ *   copyright law, including all related and neighboring rights, to the extent
+ *   allowed by law.
+ *
+ *   You can copy, modify, distribute and perform the work, even for commercial
+ *   purposes, all without asking permission.
+ */
+
+#ifndef INCLUDED_TWOM_H
+#define INCLUDED_TWOM_H
+
+#include <stdlib.h>
+#include <stdint.h>
+
+struct twom_db;
+struct twom_txn;
+struct twom_cursor;
+
+enum twom_ret {
+    TWOM_OK = 0,
+    TWOM_DONE = 1,
+    TWOM_IOERROR = -1,
+    TWOM_AGAIN = -2,
+    TWOM_EXISTS = -3,
+    TWOM_INTERNAL = -4,
+    TWOM_NOTFOUND = -5,
+    TWOM_LOCKED = -6,
+    TWOM_READONLY = -9,
+};
+
+// we don't reuse flags, there's 32 bits of space, and we spread through them
```

Comment:
I stumbled over "reuse" here. Do you mean, each flag has one and only one meaning, irrespective if other flags are set?

Reply:
They're passed to different functions, so you could have "enum twom_open_flags", "enum twom_store_flags", "enum twom_foreach_flags" etc, but the values are all disjoint so you can't have something mis-interpreted by passing it to the wrong function.

Reply:
Thanks, that explains this. If they are disjoint, you might want to organize them in the declaration such that each set of flags is visually separated from the other, e.g. 

    ...

     // repack flags
    TWOM_ALWAYSYIELD     = 1<<9,
    TWOM_NOYIELD     = 1<<10,

    // fetch flags
    ...